crates/vm/levm/src/opcode_handlers/stack_memory_storage_flow.rs

```diff
@@ -185,7 +189,9 @@ impl VM {
         let mut base_dynamic_gas: U256 = U256::zero();
 
         let storage_slot = if self.cache.is_slot_cached(&address, key) {
-            self.cache.get_storage_slot(address, key).unwrap()
+            self.cache
+                .get_storage_slot(address, key)
+                .ok_or(VMError::FatalUnwrap)?
```

Comment:
I refactored it by directly using `self.get_storage_slot`, if it is in cache it will get it from there, otherwise it will get it from db and then cache it.